src/plugins/vis_type_xy/public/utils/render_all_series.tsx

```diff
@@ -0,0 +1,197 @@
+  if (xAspect.accessor) {
+    if (xAspect.aggType === 'date_range' && xAspect.formatter) {
+      const formatter = xAspect.formatter;
+      const accessor = xAspect.accessor;
+      return (d) => {
+        const v = d[accessor];
+        if (!v) {
+          return;
+        }
+        const f = formatter(v);
+        return f;
+      };
+    }
+
+    return xAspect.accessor;
+  }
+
+  return () => (xAspect.params as FakeParams)?.defaultValue;
+};
+
+/**
+ * Renders chart Line, Area or Bar series
+ * @param config
+ * @param seriesParams
+ * @param data
+ * @param getSeriesName
+ * @param getSeriesColor
+ */
+export const renderAllSeries = (
+  { aspects, yAxes, xAxis, showValueLabel, enableHistogramMode, fittingFunction }: VisConfig,
+  seriesParams: SeriesParam[],
+  data: DatatableRow[],
+  getSeriesName: (series: XYChartSeriesIdentifier) => SeriesName,
+  getSeriesColor: SeriesColorAccessorFn,
+  timeZone: string
+) => {
+  const xAccessor = getXAccessor(aspects.x);
+
+  return seriesParams.map(
+    ({
+      show,
+      valueAxis: groupId,
+      data: { id: paramId },
+      lineWidth: strokeWidth,
+      showCircles,
+      drawLinesBetweenPoints,
+      mode,
+      interpolate,
+      type,
+    }) => {
+      const yAspect = aspects.y.find(({ aggId }) => aggId === paramId);
+
+      if (!show || !yAspect || yAspect.accessor === null) {
+        return null;
+      }
+
+      const id = `${type}-${yAspect.accessor}`;
+
+      const splitSeriesAccessors = aspects.series
+        ? (compact(aspects.series.map(({ accessor }) => accessor)) as string[])
+        : [];
+      const yAxisScale = yAxes.find(({ groupId: axisGroupId }) => axisGroupId === groupId)?.scale;
+      const stackAccessors =
+        mode === 'stacked' || yAxisScale?.mode === 'percentage' ? ['__any_value__'] : undefined;
+      const stackMode = yAxisScale?.mode === 'normal' ? undefined : yAxisScale?.mode;
+
+      switch (type) {
+        case ChartType.Histogram:
+          return (
+            <BarSeries
+              key={id}
+              id={id}
+              name={getSeriesName}
+              color={getSeriesColor}
+              tickFormat={yAspect.formatter}
+              groupId={groupId}
+              xScaleType={xAxis.scale.type}
+              yScaleType={yAxisScale?.type}
+              xAccessor={xAccessor}
+              yAccessors={[yAspect.accessor]}
+              markSizeAccessor={aspects.z?.accessor ?? undefined}
+              splitSeriesAccessors={splitSeriesAccessors}
+              data={data}
+              timeZone={timeZone}
+              stackAccessors={stackAccessors}
+              enableHistogramMode={enableHistogramMode}
+              stackMode={stackMode}
+              minBarHeight={2}
+              displayValueSettings={{
+                showValueLabel,
+                hideClippedValue: true,
+              }}
+            />
+          );
+
+        case ChartType.Area:
+        case ChartType.Line:
```

Comment:
Yeah this was solely to allow support for stacked line charts which is not available on current elastic charts. Maybe this could be a blind option for only vislib until this feature is sunsetted.

Reply:
IIRC we decided to sunset stacked line charts and migrate those charts to stacked area charts if possible

Reply:
I think you are right but I don't think but I don't recall that being decided. Because that change would alter the visualization saved object and the user would have to revert this change manually if they did not want to see the areas as there is no way to remove the area shading in visualize. So until that is sunsetted, likely in `8.0.0` I think we keep it.

Reply:
👍